### PR TITLE
feat: add clean schedule and resolve npm audit issues

### LINKED ISSUE #25: [Feature Request] Custom clean up schedule (rate)
Would you be willing to accept a pull request to parameterize the schedule for invoking the Lambda function in your CloudFormation template that is currently hard coded to `rate(1 hour)`? (Template.yaml line 41)

The proposed change would be to add an optional parameter with the default of the current value (`rate(1 hour)`).

This would allow us to invoke it much less frequently for some scenarios where changes are less frequent.

### PR DESCRIPTION
**Pull request by afittz**
Fixes #25 

- Added new CloudFormation parameter called `CleanSchedule` with a default value of `rate(1 hour)`.

Additional changes:
- Moved `aws-sdk` npm package to devDependencies since AWS includes this in the runtime image and it increases the deployment size. (https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html)
- Resolved npm audit issues.
- Added .gitignore values for deploying via SAM.

I'm happy to change/remove anything; just let me know!

### CODE REVIEW

**Comment by afittz** [comment]
Are there any changes I need to make for this?

**Comment by theburningmonk** [comment]
@afittz sorry, missed this when it came in, looks good, maybe add a few examples in the description for the `CleanSchedule` param so ppl don't have to look at the code to figure out what they need to put in there (not everyone is familiar with the cron syntax for CW Events)

**Comment by afittz** [comment]
@theburningmonk Great suggestion! I'll add some examples.

**Comment by afittz** [comment]
@theburningmonk Sorry for the delay in getting this cleaned up!!!

File: README.md [hunk 1]
```diff
@@ -39,8 +39,11 @@ AutoDeployMyAwesomeLambdaLayer:
       SemanticVersion: <enter latest version>
     Parameters:
       VersionsToKeep: <defaults to 3>
+      CleanSchedule: <defaults to rate(1 hour)>
 ```
 
+The CleanSchedule parameter supports any Schedule Expression for a CloudWatch event. The default uses a rate expression that will execute hourly, however you can easily specify any desired number of minutes/hours/days between runs or use a cron expression. Examples: `rate(12 hours)` or `rate(1 day)`. More details on valid options can be found on the [Amazon CloudWatch Events website](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
```
Comment (JackKelly-Bellroy) [inline README.md:45]: I think they're called "EventBridge events" now

**Comment by arjunsoota** [comment]
Why was this not merged to the main release branch ?

**Pull request closed without merging**